apps/app.go

```diff
@@ -54,6 +62,33 @@ type App struct {
 	GrantedLocations Locations `json:"granted_locations,omitempty"`
 }
 
+func DecodeCompatibleApp(data []byte) (app *App, err error) {
```

Comment:
Shall we create tickets to remove these migration steps in the near future (i.e. v0.9)? I see the value and the need here, but we should try to update the current apps to the "right shape" ASAP and stop supporting v0.7 apps, among other things to keep the clode cleaner.